src/middlewared/middlewared/api/v25_04_0/smb.py

```diff
@@ -63,3 +99,65 @@ class GetSmbAclArgs(BaseModel):
 
 class GetSmbAclResult(SetSmbAclResult):
     pass
+
+
+SMBEncryption = Literal['DEFAULT', 'NEGOTIATE', 'DESIRED', 'REQUIRED']
+
+
+class SmbServiceEntry(BaseModel):
+    id: int
+    netbiosname: NetbiosName
+    """ Netbios name of this server """
+    netbiosalias: list[NetbiosName]
+    """ Alternative netbios names of the server that will be announced via
+    netbios nameserver and registered in active directory when joined."""
+    workgroup: NetbiosDomain
+    """ Workgroup. When joined to active directory, this will be automatically
+    reconfigured to match the netbios domain of the AD domain. """
+    description: str
+    """ Description of SMB server. May appear to clients during some operations. """
+    enable_smb1: bool
+    """ Enable SMB1 support for server. WARNING: using SMB1 protocol is not recommended """
+    unixcharset: SMBCharsetType
+    """ Select characterset for file names on local filesystem. This should only be used
+    in cases where system administrator knows that the filenames are not UTF-8."""
+    localmaster: bool
+    syslog: bool
+    """ Send log messages to syslog. This should be enabled if system administrator
+    wishes for SMB server error logs to be included in information sent to remote syslog
+    server if this is globally configured for TrueNAS."""
+    aapl_extensions: bool
+    """ Enable support for SMB2/3 AAPL protocol extensions. This changes the TrueNAS server
+    so that it is advertised as supporting Apple protocol extensions as a MacOS server, and
+    is required for Time Machine support. """
+    admin_group: str | None
+    """ The selected group will have full administrator privileges on TrueNAS over SMB protocol. """
+    guest: NonEmptyString
+    filemask: UnixPerm | Literal['DEFAULT']
+    """ smb.conf create mask. DEFAULT applies current server default which is 664. """
+    dirmask: UnixPerm | Literal['DEFAULT']
+    """ smb.conf directory mask. DEFAULT applies current server default which is 775. """
+    ntlmv1_auth: bool
+    """ Enable legacy and very insecure NTLMv1 authentication. This should never be done except
+    in extreme edge cases and may be against regulations in non-home environments. """
+    multichannel: bool
+    encryption: SMBEncryption
+    bindip: list[IPvAnyInterface]
+    server_sid: SID | None
+    """ Universally-unique identifier for this particular SMB server that serves as domain SID
+    for all local SMB user and group accounts """
+    smb_options: str
+    """ Additional unvalidated and unsupported configuration options for the SMB server.
+    WARNING: using smb_options may produce unexpected server behavior. """
+    debug: bool
```

Comment:
Maybe default `debug` to False?

Reply:
Since we're dealing with update method its omission leads to it being left as-is. This means that it hits the datastore default which is set to False in migration.